src/gui/Src/Gui/HandlesView.cpp

```diff
@@ -0,0 +1,432 @@
+            char typeStr[512];
+            memset(name, 0, sizeof(name));
+            memset(typeStr, 0, sizeof(typeStr));
+            DbgGetHandleName(name, typeStr, sizeof(name), allHandles[i]);
+            mHandlesTable->setCellContent(i, 0, typeStr);
+            mHandlesTable->setCellContent(i, 1, ToHexString(typeNumbers[i]));
+            mHandlesTable->setCellContent(i, 2, ToHexString(allHandles[i]));
+            mHandlesTable->setCellContent(i, 3, ToHexString(grantedAccess[i]));
+            mHandlesTable->setCellContent(i, 4, name);
+        }
+        delete allHandles;
+        delete typeNumbers;
+        delete grantedAccess;
+    }
+    else
+    {
+        mHandlesTable->setRowCount(1);
+        mHandlesTable->setCellContent(0, 0, QString("error:%1").arg(handleCount));
+    }
+    mHandlesTable->reloadData();
+}
+
+void HandlesView::enumPrivileges()
+{
+    mPrivilegesTable->setRowCount(35);
+    AppendPrivilege(0, "SeAssignPrimaryTokenPrivilege");
+    AppendPrivilege(1, "SeAuditPrivilege");
+    AppendPrivilege(2, "SeBackupPrivilege");
+    AppendPrivilege(3, "SeChangeNotifyPrivilege");
+    AppendPrivilege(4, "SeCreateGlobalPrivilege");
+    AppendPrivilege(5, "SeCreatePagefilePrivilege");
+    AppendPrivilege(6, "SeCreatePermanentPrivilege");
+    AppendPrivilege(7, "SeCreateSymbolicLinkPrivilege");
+    AppendPrivilege(8, "SeCreateTokenPrivilege");
+    AppendPrivilege(9, "SeDebugPrivilege");
+    AppendPrivilege(10, "SeEnableDelegationPrivilege");
+    AppendPrivilege(11, "SeImpersonatePrivilege");
+    AppendPrivilege(12, "SeIncreaseBasePriorityPrivilege");
+    AppendPrivilege(13, "SeIncreaseQuotaPrivilege");
+    AppendPrivilege(14, "SeIncreaseWorkingSetPrivilege");
+    AppendPrivilege(15, "SeLoadDriverPrivilege");
+    AppendPrivilege(16, "SeLockMemoryPrivilege");
+    AppendPrivilege(17, "SeMachineAccountPrivilege");
+    AppendPrivilege(18, "SeManageVolumePrivilege");
+    AppendPrivilege(19, "SeProfileSingleProcessPrivilege");
+    AppendPrivilege(20, "SeRelabelPrivilege");
+    AppendPrivilege(21, "SeRemoteShutdownPrivilege");
+    AppendPrivilege(22, "SeRestorePrivilege");
+    AppendPrivilege(23, "SeSecurityPrivilege");
+    AppendPrivilege(24, "SeShutdownPrivilege");
+    AppendPrivilege(25, "SeSyncAgentPrivilege");
+    AppendPrivilege(26, "SeSystemEnvironmentPrivilege");
+    AppendPrivilege(27, "SeSystemProfilePrivilege");
+    AppendPrivilege(28, "SeSystemtimePrivilege");
+    AppendPrivilege(29, "SeTakeOwnershipPrivilege");
+    AppendPrivilege(30, "SeTcbPrivilege");
+    AppendPrivilege(31, "SeTimeZonePrivilege");
+    AppendPrivilege(32, "SeTrustedCredManAccessPrivilege");
+    AppendPrivilege(33, "SeUndockPrivilege");
+    AppendPrivilege(34, "SeUnsolicitedInputPrivilege");
+    mPrivilegesTable->reloadData();
+}
+
+void HandlesView::AppendPrivilege(int row, const char* PrivilegeString)
+{
+    DbgCmdExec(QString("GetPrivilegeState \"%1\"").arg(PrivilegeString).toUtf8().constData());
```

Comment:
This can return before the command finished. You should use `DbgCmdExecDirect` in this case probably.